File: lib/features/planning/domain/repositories/planning_repository.dart
import 'package:dartz/dartz.dart';
import '../../../../core/errors/failures.dart';
import '../entities/phase_entity.dart';
import '../entities/planning_task_entity.dart';

abstract class PlanningRepository {
  // Phases
  Future<Either<Failure, List<PhaseEntity>>> getPhasesByProject(String projectId);
  Future<Either<Failure, PhaseEntity>> createPhase(PhaseEntity phase);
  Future<Either<Failure, PhaseEntity>> updatePhase(PhaseEntity phase);
  Future<Either<Failure, void>> deletePhase(String id);

  // Tâches de planning
  Future<Either<Failure, List<PlanningTaskEntity>>> getTasksByPhase(String phaseId);
  Future<Either<Failure, List<PlanningTaskEntity>>> getTasksByProject(String projectId);
  Future<Either<Failure, PlanningTaskEntity>> getPlanningTaskById(String id);
  Future<Either<Failure, PlanningTaskEntity>> createPlanningTask(PlanningTaskEntity task);
  Future<Either<Failure, PlanningTaskEntity>> updatePlanningTask(PlanningTaskEntity task);
  Future<Either<Failure, void>> deletePlanningTask(String id);
}

File: src/features/planning/domain/repositories/planning_repository.ts
/**
 * AGB CHANTIER - Interface du Repository Planning & Gantt - AXE 06
 */

import { PhaseEntity, PlanningStats } from "../entities/planning_entity";

export interface PlanningRepository {
  getAllPhases(): Promise<PhaseEntity[]>;
  getPhasesByProject(projectId: string): Promise<PhaseEntity[]>;
  getPhaseById(id: string): Promise<PhaseEntity | null>;
  createPhase(phase: Omit<PhaseEntity, "id" | "createdAt" | "updatedAt">): Promise<PhaseEntity>;
  updatePhase(phase: PhaseEntity): Promise<PhaseEntity>;
  deletePhase(id: string): Promise<void>;
  updatePhaseProgress(id: string, progress: number, status?: PhaseEntity["status"]): Promise<PhaseEntity>;
  toggleMilestone(phaseId: string, milestoneId: string, isReached: boolean): Promise<PhaseEntity>;
  calculatePlanningStats(projectId?: string): Promise<PlanningStats>;
  seedInitialPhasesIfEmpty(): Promise<void>;
}
